pkg/security/security.go

```diff
@@ -235,3 +244,14 @@ type SecretItem struct {
 
 	ExpireTime time.Time
 }
+
+type CredFetcher interface {
+	// GetPlatformCredential fetches workload credential provided by the platform.
+	GetPlatformCredential() (string, error)
+
+	// GetType returns credential fetcher type. Supported types include "Kubernetes", "GoogleComputeEngine", "Mock".
+	GetType() string
+
+	// The name of the IdentityProvider that can authenticate the workload credential.
+	GetIdentityProvider() string
```

Comment:
Its not clear to me how this should be used, and what value a credential fetcher should provide here

Reply:
This the IDP that issued the workload credential. CA needs this information to authenticate workload credential before issuing certificate to the workload.

Reply:
That's not even possible to support in most cases. What if the token is OAuth2 access ? There is no information about who the issuer is. And it shouldn't be something we care about - the gRPC interface has the proper abstraction, it returns the set of headers that need to be added to request, without requiring that the implementation knows the actual IDP. 

Also IDP is usually an URL - not clear from the comments.

Reply:
issuer-uri is just one type of identity providers. For example, if we support OAuth2 token at one point, it may be the token endpoint. IdentityProvider has to be registered to CA, so that CA can authenticate the credential.

Reply:
That's something the fetcher will handle internally - why do we care ? Same for GetType - in reality all we need from a token fetcher is what gRPC interface defines, returning the set of headers that will need to be set in the request. 
Different tokens use different header types - and may use other headers besides Authorization. But that's opaque, we can't require that all token providers have ( or expose ) the identity provider URL.